config.go

```diff
@@ -161,58 +142,30 @@ func (mvc *MinValuableConfig) ApplyEnv() {
 	}
 }
 
-// todo: merge it with (ca *MinValuableConfig) ApplyEnv() when config separated from Cagent
-func (ca *Cagent) ApplyEnv() {
-	if val, ok := os.LookupEnv("CAGENT_HUB_URL"); ok {
-		ca.HubURL = val
-	}
-
-	if val, ok := os.LookupEnv("CAGENT_HUB_USER"); ok {
-		ca.HubUser = val
-	}
-
-	if val, ok := os.LookupEnv("CAGENT_HUB_PASSWORD"); ok {
-		ca.HubPassword = val
-	}
-}
-
-func (ca *Cagent) SetVersion(version string) {
-	ca.version = version
-}
-
-func (ca *Cagent) userAgent() string {
-	if ca.version == "" {
-		ca.version = "{undefined}"
-	}
-	parts := strings.Split(ca.version, "-")
-
-	return fmt.Sprintf("Cagent v%s %s %s", parts[0], runtime.GOOS, runtime.GOARCH)
-}
-
-func (ca *Cagent) DumpConfigToml() string {
+func (cfg *Config) DumpToml() string {
 	buff := &bytes.Buffer{}
 
-	err := toml.NewEncoder(buff).Encode(ca)
+	err := toml.NewEncoder(buff).Encode(cfg)
 	if err != nil {
-		log.Errorf("DumpConfigToml error: %s", err.Error())
+		log.Errorf("DumpToml error: %s", err.Error())
 		return ""
 	}
 
 	return buff.String()
 }
 
-func (ca *Cagent) ReadConfigFromFile(configFilePath string) error {
+func (cfg *Config) ReadFromFile(configFilePath string) error {
```

Comment:
It overwrittes only if field is present in the file.
Signature change makes sense

Reply:
That might reasonable to rename to something `TryUpgradeFromFile`

Reply:
So it may be better to rename it to `FillFromFile`?

Reply:
👍

Reply:
So, _when_ exactly does it overwrite?
If the field is already set, it will be overwritten and otherwise it will be set right? Which basically is overwriting the `nil` value of the field. So basically the same as when it's set or not because every value, except pointers, has a default value.
`FillFromFile`  doesn't tell _me_ immediately what will happen. `Fill` is quite unspecific. `TryUpgradeFromFile` is more clear IMHO.
But I remember again why it was attached to `*Config`. We first have the default config, then overwrite from config file and then apply env and flags. I forgot about that. What's your opinion on `func UpdateConfigFromFile(cfg *Config, configFilePath string) err`